### PR TITLE
fix: [M3-8802] - Object Storage landing page UI bugs

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Fixes numerous UI bugs 🐛  on the Object Storage landings pages (Buckets 🪣  and Access Keys 🔑)
- Does small clean up and sorting of props along the way 🧹 

## Changes with Preview 📷

<details>
  <summary>Fixed inconstant table background color on the Buckets landing page</summary>
  
  | Before  | After   |
  | ------- | ------- |
  | Screenshot 2024-10-29 at 3 00 11 PM | Screenshot 2024-10-29 at 3 00 03 PM |
</details>

<details>
  <summary>Fixed broken loading state on Access Key landing page </summary>
  
  | Before  | After   |
  | ------- | ------- |
  | Screenshot 2024-10-29 at 3 01 54 PM | Screenshot 2024-10-29 at 3 02 06 PM |
</details>

<details>
  <summary>Improved Access Keys landing page for small screen sizes 📱</summary>
  
  I hid the "Hostnames" column on small screens, **but** added an action menu item so the user can still open and view the hostnames drawer
  
  | Before  | After   |
  | ------- | ------- |
  | Screenshot 2024-10-29 at 3 11 02 PM | Screenshot 2024-10-29 at 3 10 53 PM |
</details>

<details>
  <summary>Removed "Actions" text in table column header</summary>
  
  | Before  | After   |
  | ------- | ------- |
  | Screenshot 2024-10-29 at 3 12 24 PM | Screenshot 2024-10-29 at 3 12 18 PM |
</details>

<details>
  <summary>Fixed broken action menu spacing</summary>
  
  | Before  | After   |
  | ------- | ------- |
  | Screenshot 2024-10-29 at 3 14 05 PM | Screenshot 2024-10-29 at 3 13 54 PM |
</details>

<details>
  <summary>Fixed multiple background colors in hostnames drawer</summary>
  
  | Before  | After   |
  | ------- | ------- |
  | Screenshot 2024-10-29 at 3 17 04 PM | Screenshot 2024-10-29 at 3 17 12 PM |
</details>

## How to test 🧪

- First, make sure you have some Buckets 🪣  and Access Keys 🔑 on your account
- Check http://localhost:3000/object-storage/access-keys and http://localhost:3000/object-storage/buckets
  - Verify styles are better than or equal to the current UI's styles 🎨 
  - Verify there are no regressions

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by hana-akamai — COMMENTED** [review]
Should there be a min label column width for desktop view? The column feels a bit cramped now

image

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyTableBody.tsx [hunk 1]
```diff
@@ -1,73 +1,79 @@
...
-      )}
-    </>
-  );
+  if (data?.length === 0) {
+    return <TableRowEmpty colSpan={cols} />;
+  }
+
+  return data?.map((eachKey: ObjectStorageKey, index) => (
+    <AccessKeyTableRow
+      key={index}
```
Comment (hana-akamai) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyTableBody.tsx:71]:
```suggestion
      key={eachKey.id}
```

**Review by hana-akamai — APPROVED** [review]

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyTableBody.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyTableBody.tsx:71]: Great catch. Thank you!

**Comment by mjac0bs** [comment]
👋🏼  It's me, hi, my [PR](https://github.com/linode/manager/pull/11143) just caused merge conflict problems. 

Thanks for resolving and I'll review this PR in after I take lunch. Looks like solid clean up!

**Review by mjac0bs — APPROVED** [review]
This looked really good! Styling at various screen sizes checked out and no regressions that I could find in functionality. Great attention to detail here - I don't know that I would have noticed the different background colors in hostnames copyable text field on my own. 🚀

**Pull request merged**